app/src/main/java/com/google/droidjump/LevelsFragment.java

```diff
@@ -16,26 +16,70 @@
 
 package com.google.droidjump;
 
-import static androidx.navigation.Navigation.findNavController;
+import android.annotation.SuppressLint;
+import android.content.Context;
 import android.os.Bundle;
+import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.Button;
+import android.widget.GridView;
+import android.widget.ImageButton;
+import androidx.navigation.Navigation;
+import java.util.Objects;
 
 /**
  * Displays Levels Screen.
  */
 public class LevelsFragment extends Fragment {
 
+    private MainActivity activity;
+    private int[] levels;
+    private LevelsAdapter adapter;
+
     @Override
-    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        View rootView = inflater.inflate(R.layout.levels_screen, container, /* attachToRoot= */ false);
-        Button levelPlayButton = rootView.findViewById(R.id.level_play);
-        levelPlayButton.setOnClickListener(view -> {
-            findNavController(view).navigate(R.id.action_levels_screen_to_game_screen);
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        activity = Objects.requireNonNull((MainActivity) getActivity());
+        levels = createLevels(activity.getLevelsCount());
+        Context context = getContext();
+        adapter = new LevelsAdapter(Objects.requireNonNull(context), levels,
+                activity.getCurrentLevel());
+    }
+
+    @SuppressLint("SetTextI18n")
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.levels_screen, container,
+                /* attachToRoot= */ false);
+
+        // Finding gridView and putting levels to it.
+        GridView gridView = rootView.findViewById(R.id.levels_grid_view);
+        gridView.setAdapter(Objects.requireNonNull(adapter));
+
+        // Adding onClick events.
+        gridView.setOnItemClickListener((adapterView, view, i, l) -> {
+            if (activity.getCurrentLevel() >= (int) adapter.getItem(i)) {
+                Navigation.findNavController(view).navigate(
+                        R.id.action_levels_screen_to_game_screen);
+            }
+        });
+
+        // Redirecting on click to a start screen.
+        ImageButton menuButton = rootView.findViewById(R.id.menu_button);
+        menuButton.setOnClickListener(view -> {
+            activity.onBackPressed();
         });
         return rootView;
     }
+
+    private int[] createLevels(int levelsCount) {
```

Comment:
I would suggest to create Level class instead and return List<Level>. Level right now might contain only it's order number inside, I believe Daria would add more later.

Reply:
Real implementation won't use "levelsCount" - I believe it is only needed for now to generate fake levels. Don't pass it inside the method and use "activity.getLevelsCount()" right from here.

Also, I guess "loadLevels" is better than "createLevels" since real implementation won't create them, just load from text files.

Reply:
1. I am not sure that we need to have a list of all Level objects because we need just their numbers without their data. (Probably when we create a level, it parses data from JSON in a constructor). It's quicker to create an array of integers using the levelsCount variable than create a Level object + parse data from JSON, then add it to the list.
2. Why not? We can extract a count of levels using this line of code: R.raw.class.getFields().length; (raw - levels folder)
3. Yes, sure. Will change it.